### PR TITLE
feat(hfx): option to suppress sending 500's on error

### PR DESCRIPTION
**Pull request by Bobby-McBobface**
Discussed on Discord
https://discord.com/channels/254360814063058944/254360814063058944/1126380782055862366

### CODE REVIEW

**Review by favna — DISMISSED** [review]

File: packages/http-framework/src/lib/Client.ts [hunk 1]
```diff
@@ -151,6 +153,11 @@ export interface ClientOptions {
 	 * @default `1024 * 1024` (1 MiB)
 	 */
 	bodySizeLimit?: number;
+	/**
```
Comment (kyranet) [inline packages/http-framework/src/lib/Client.ts:156]:
```suggestion
	bodySizeLimit?: number;

	/**
```

File: packages/http-framework/src/lib/Client.ts [hunk 2]
```diff
@@ -151,6 +153,11 @@ export interface ClientOptions {
...
+	 * Whether to suppress automatically sending a 500 status to Discord if errors occur in an interaction.
```
Comment (kyranet) [inline packages/http-framework/src/lib/Client.ts:157]:
```suggestion
	 * Whether to send a 500 status to Discord if an error occurs while processing an interaction.
```

File: packages/http-framework/src/lib/interactions/utils/util.ts [hunk 3]
```diff
@@ -106,6 +106,7 @@ export type TransformRaw<T extends BaseInteractionType> = T extends Autocomplete
 export function handleError(response: ServerResponse, error: unknown): ServerResponse {
 	container.client.emit('error', error);
 
+	if (container.client.suppressErrorHandling) return response;
 	if (response.closed) return response;
```
Comment (kyranet) [inline packages/http-framework/src/lib/interactions/utils/util.ts:110]:
```suggestion
	if (container.client.suppressErrorHandling || response.closed) return response;
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: packages/http-framework/src/lib/Client.ts (hunk 2 above)
Comment (Bobby-McBobface) [reply packages/http-framework/src/lib/Client.ts:158]: That changes the meaning of the sentence. The option is `true` if the error is suppressed, i.e. no 500 is sent.
Perhaps I could rename the variable to `handleErrors` or something so it's not a negating statement?

Comment (kyranet) [reply packages/http-framework/src/lib/Client.ts:158]: Yes, please. I think a more fitting name would be `replyOnError` or maybe `httpReplyOnError`. 🤔

**Review by favna — APPROVED** [review]

**Review by SicklineBro7 — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**